src/Containers/Microsoft.NET.Build.Containers/LocalDaemons/DockerCli.cs

```diff
@@ -196,17 +205,68 @@ internal static JsonDocument GetDockerConfig()
             }
 
             return JsonDocument.Parse(dockerCommandResult.StdOut);
-
-
         }
         catch (Exception e) when (e is not DockerLoadException)
         {
             throw new DockerLoadException(Resource.FormatString(nameof(Strings.DockerInfoFailed_Ex), e.Message));
         }
     }
+    /// <summary>
+    /// Checks if the registry is marked as insecure in the docker/podman config.
+    /// </summary>
+    /// <param name="registryDomain"></param>
+    /// <returns></returns>
+    public static bool IsInsecureRegistry(string registryDomain)
+    {
+        try
+        {
+            //check the docker config to see if the registry is marked as insecure
+            var rootElement = GetDockerConfig().RootElement;
+
+            //for docker
+            if (rootElement.TryGetProperty("RegistryConfig", out var registryConfig) && registryConfig.ValueKind == JsonValueKind.Object)
+            {
+                if (registryConfig.TryGetProperty("IndexConfigs", out var indexConfigs) && indexConfigs.ValueKind == JsonValueKind.Object)
+                {
+                    foreach (var property in indexConfigs.EnumerateObject())
+                    {
+                        if (property.Value.ValueKind == JsonValueKind.Object && property.Value.TryGetProperty("Secure", out var secure) && !secure.GetBoolean())
+                        {
+                            if (property.Name.Equals(registryDomain, StringComparison.Ordinal))
```

Comment:
nit: Overall looks good, but I would propose to refactor some nested `if`s. For example extracting  expressions to local variables and the use single `if`. It will reduce cyclomatic complexity and increase readability.